### PR TITLE
Transaction implementation

### PR DESCRIPTION
**Pull request by getbez**

### CODE REVIEW

File: app/models/dispatch.rb [hunk 1]
```diff
@@ -24,9 +24,15 @@
...
+
     acts_as_paranoid 
 
     belongs_to :fdp 
     
     has_many :dispatch_items
+
+    after_save :pre_post
+
+    after_update :reverse
```
Comment (asuraphel) [inline app/models/dispatch.rb:37]: I think we should do `:patch` here instead. Which is reversal followed by a new `Posting`.

**Review by asuraphel — CHANGES_REQUESTED** [review]
Awesome work! I just have one comment.

File: app/models/dispatch.rb (hunk 1 above)
Comment (getbez) [reply app/models/dispatch.rb:37]: are you suggesting a name change? I'm fine with that but in this case actually after_update returns save's active transaction so it means reverse then post are executed. reverse only does reverse and then save is called to do the posting.

Comment (getbez) [reply app/models/dispatch.rb:37]: I actually was going to change the :prepost to :post  but i have another post method which actually does the posting. that'll soon be fixed.

**Pull request merged**